catalog/spatial/kml/spatial-kml-util/src/main/java/org/codice/ddf/spatial/kml/converter/MetacardToKml.java

```diff
@@ -0,0 +1,142 @@
+          "WKT was null or empty. Unable to preform KML Transform on Metacard.");
+    }
+
+    com.vividsolutions.jts.geom.Geometry geo = readGeoFromWkt(wkt);
+    Geometry kmlGeo = createKmlGeo(geo);
+    if (!POINT_TYPE.equals(geo.getGeometryType())) {
+      kmlGeo = addPointToKmlGeo(kmlGeo, geo.getCoordinate());
+    }
+    return kmlGeo;
+  }
+
+  private static Geometry createKmlGeo(com.vividsolutions.jts.geom.Geometry geo)
+      throws CatalogTransformerException {
+    Geometry kmlGeo;
+    if (POINT_TYPE.equals(geo.getGeometryType())) {
+      kmlGeo = createPointGeo((Point) geo);
+    } else if (LINES_STRING_TYPE.equals(geo.getGeometryType())) {
+      kmlGeo = createLineStringGeo((LineString) geo);
+    } else if (POLYGON_TYPE.equals(geo.getGeometryType())) {
+      kmlGeo = createPolygonGeo((Polygon) geo);
+    } else if (geo instanceof GeometryCollection) {
+      kmlGeo = createGeometryCollectionGeo(geo);
+    } else {
+      throw new CatalogTransformerException(
+          "Unknown / Unsupported Geometry Type '"
+              + geo.getGeometryType()
+              + "'. Unale to preform KML Transform.");
+    }
+    return kmlGeo;
+  }
+
+  private static Geometry createGeometryCollectionGeo(com.vividsolutions.jts.geom.Geometry geo)
+      throws CatalogTransformerException {
+    Geometry kmlGeo;
+    List<Geometry> geos = new ArrayList<>();
+    for (int xx = 0; xx < geo.getNumGeometries(); xx++) {
+      geos.add(createKmlGeo(geo.getGeometryN(xx)));
+    }
+    kmlGeo = KmlFactory.createMultiGeometry().withGeometry(geos);
+    return kmlGeo;
+  }
+
+  private static Geometry createPolygonGeo(Polygon geo) {
+    Geometry kmlGeo;
+    Polygon jtsPoly = geo;
+    de.micromata.opengis.kml.v_2_2_0.Polygon kmlPoly = KmlFactory.createPolygon();
+    List<Coordinate> kmlCoords =
+        kmlPoly.createAndSetOuterBoundaryIs().createAndSetLinearRing().createAndSetCoordinates();
+    for (com.vividsolutions.jts.geom.Coordinate coord : jtsPoly.getCoordinates()) {
+      kmlCoords.add(new Coordinate(coord.x, coord.y));
+    }
+    kmlGeo = kmlPoly;
+    return kmlGeo;
+  }
+
+  private static Geometry createLineStringGeo(LineString geo) {
+    Geometry kmlGeo;
+    LineString jtsLS = geo;
+    de.micromata.opengis.kml.v_2_2_0.LineString kmlLS = KmlFactory.createLineString();
+    List<Coordinate> kmlCoords = kmlLS.createAndSetCoordinates();
+    for (com.vividsolutions.jts.geom.Coordinate coord : jtsLS.getCoordinates()) {
+      kmlCoords.add(new Coordinate(coord.x, coord.y));
+    }
+    kmlGeo = kmlLS;
+    return kmlGeo;
+  }
+
+  private static Geometry createPointGeo(Point geo) {
+    Geometry kmlGeo;
+    Point jtsPoint = geo;
+    kmlGeo = KmlFactory.createPoint().addToCoordinates(jtsPoint.getX(), jtsPoint.getY());
+    return kmlGeo;
+  }
+
+  private static com.vividsolutions.jts.geom.Geometry readGeoFromWkt(final String wkt)
+      throws CatalogTransformerException {
+    WKTReader reader = new WKTReader();
```

Comment:
We can use `ThreadLocal` to avoid having to instantiate this on every call

Reply:
Added.